python/ray/_private/gcs_utils.py

```diff
@@ -119,6 +121,57 @@ def create_gcs_channel(address: str, aio=False):
     return init_grpc_channel(address, options=_GRPC_OPTIONS, asynchronous=aio)
 
 
+def _auto_reconnect(f):
+    @wraps(f)
+    def wrapper(self, *args, **kwargs):
+        while True:
+            try:
+                return f(self, *args, **kwargs)
+            except grpc.RpcError as e:
+                if self._enable_auto_reconnect is False:
+                    raise
+                if e.code() == grpc.StatusCode.UNAVAILABLE:
+                    logger.error(
+                        "Failed to send request to gcs, reconnecting. "
+                        f"Error {e}")
+                    try:
+                        self._connect()
+                    except Exception:
+                        logger.error(f"Connecting to gcs failed. Error {e}")
+                    time.sleep(1)
```

Comment:
Could we have some max retry backoff (e.g., 10 times)?

Reply:
But in raylet, we actually keep retrying this because you don't know when will gcs get reconnected. I actually have auto_reconnect flags to enable/disable this feature.
How about this, I put a number there in parameter, and if it's set, there is a limit retry.